Add support for WMTS capabilities

spatialillusions opened the pull request:
This will implement #274 and add support for WMTS Capabilites without needing to connect to wmts.maptiler.com. I have tested the code in QGIS to verify that I am able to connect to the services. 

Please provide feedback if it is anything you want me to change.

spatialillusions commented:
Sorry for the outdated template, I did my development in a virtual machine and forgot to move out the latest version of the template. Should be correct now.

spatialillusions commented:
Would be very nice if @klokan or @petrsloup could give me a timeframe for when/if this will be merged. Thanks!

petrsloup commented on src/server.js:
```diff
@@ -325,11 +326,6 @@ function start(opts) {
 
-        var query = req.query.key ? ('?key=' + req.query.key) : '';
-        data_.wmts_link = 'http://wmts.maptiler.com/' +
-          base64url('http://' + req.headers.host +
```
Was this the only reference to the `base64url` package? I guess we could remove the dependency completely...

petrsloup commented on src/server.js:
```diff
@@ -374,6 +370,20 @@ function start(opts) {
+    if (!wmts) {
+      return null;
+    }
+    if(wmts.hasOwnProperty("serve_rendered")&!wmts.serve_rendered){
```
Please fix code formatting here (space after `if`, `&` -> ` && ` (with spaces around), `){` -> `) {`, indentation of the return below, ...)

spatialillusions commented:
@petrsloup Should I remove base64url from package.json as well? As far as I can tell it is not used in any other places in the code.

petrsloup commented:
Yes, please remove the dependency completely -- it's not used anywhere else in the project.

petrsloup commented:
@spatialillusions https://github.com/klokantech/tileserver-gl/blob/e8b3c0254678f049bc1ef67bc44cd5add4aa81da/src/server.js#L300 This should be probably also removed, right?

spatialillusions commented:
Removed the unused variables, and also added the key to the template so that it should work for your hosted services. (Not sure if the key is used for any other links in the template.)

petrsloup commented:
Thanks, I don't see any more issues and everything seems to be working now. Merging...
